libethereum/EthereumHost.cpp

```diff
@@ -576,8 +576,20 @@ void EthereumHost::onPeerNewBlock(EthereumPeer* _peer, RLP const& _r)
 				u256 difficulty = _r[1].toInt<u256>();
 				if (m_syncingTotalDifficulty < difficulty)
 				{
+					BlockInfo bi;
+					RLP root(_r[0].data());
+					try
+					{
+						bi.populateFromHeader(root[0], CheckEverything, h);
```

Comment:
The difficulty on line 577 comes from the packet and this check can be bypassed with ease. +1 for tracking bad blocks origin and banning peers that sent them